FreeTubeApp/FreeTube
# Switch from sync fs APIs to fs/promises

# Switch from sync fs APIs to fs/promises




## Pull Request Type


## Related issue
related to #1115

## Description
Currently we use the synchronous file system APIs for the majority of the filesystem operations, these are blocking, which is problematic as JavaScript is single threaded, so they prevent any other code from running while the operations are happening. Migrating to the promises file system APIs should alleviate that issue.

## Testing 
1. Enable and disable the experimental replace HTTP cache setting and check that the file is created and removed properly in the userdata directory
2. Change the language, check that the new one is loaded and that the region names for trending show up
3. Watch a video through the local API and check that the storyboard and DASH files are created and removed correctly when you change to a different page
4. Export something in the data settings, checking that the file is written properly to your chosen export location
5. Import something in the data settings
6. Download a video inside FreeTube (downloading an audio track will be the fastest to test), check that the file is correctly written to your download location

## Desktop

- **OS:** Windows
- **OS Version:** 10
- **FreeTube version:** 0.18.0
